File: frontend/src/api/goldPrice.ts
import request from '@/utils/request'

export interface GoldPrice {
  price: number
  changeAmount: number
  changePercent: number
  currency: string
  symbol: string
  timestamp: string
  high: number
  low: number
  average: number
  volatility: number
}

export interface PricePoint {
  date: string
  price: number
}

export interface Currency {
  code: string
  name: string
  symbol: string
  flag: string
  rate: number
}

export const goldPriceApi = {
  getCurrentPrice: (currency = 'CNY') =>
    request.get(`/gold-price/current?currency=${currency}`),
  getPriceHistory: (currency = 'CNY', days = 30) =>
    request.get(`/gold-price/history?currency=${currency}&days=${days}`),
  getSupportedCurrencies: () => request.get('/gold-price/currencies'),
  refresh: () => request.post('/gold-price/refresh'),
}

File: api/app/schemas/gold_price.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class GoldPriceDTO(BaseModel):
    price: float
    change_amount: float | None = Field(None, serialization_alias="changeAmount")
    change_percent: float | None = Field(None, serialization_alias="changePercent")
    currency: str
    symbol: str
    timestamp: datetime | None = None
    high: float | None = None
    low: float | None = None
    average: float | None = None
    volatility: float | None = None

    model_config = ConfigDict(populate_by_name=True, by_alias=True)


class PricePoint(BaseModel):
    date: str
    price: float


class CurrencyDTO(BaseModel):
    code: str
    name: str
    symbol: str
    flag: str
    rate: float

    model_config = ConfigDict(populate_by_name=True, by_alias=True)
